src/Spatial.Tests/Euclidean/PlaneTest.cs

```diff
@@ -80,6 +85,77 @@ public void ProjectPointOn(string ps, string rootPoint, string unitVector, strin
             AssertGeometry.AreEqual(expected, projectedPoint, eps);
         }
 
+        [TestCase(ZeroPoint, X, Y)] //all 3 points are on the plane z=0. normal vector is +ez.
+        [TestCase(ZeroPoint, Y, Z)] //all 3 points are on the plane x=0. normal vector is +ex.
+        [TestCase(ZeroPoint, Z, X)] //all 3 points are on the plane y=0. normal vector is +ey.
+        public void CreateFittedPlaneFrom_PointsShouldBeOnTheFittedPlane(string sP1, string sP2, string sP3)
+        {
+            var ps = new[]
+            {
+                Point3D.Parse(sP1),
+                Point3D.Parse(sP2),
+                Point3D.Parse(sP3),
+            }.ToArray();
+            var aPlane = Plane.CreateFittedPlaneFrom(ps);
+
+            var tolerance = 1e-16;
+            Assert.That(aPlane.SignedDistanceTo(ps[0]), Is.EqualTo(0).Within(tolerance), "ps[0]");
+            Assert.That(aPlane.SignedDistanceTo(ps[1]), Is.EqualTo(0).Within(tolerance), "ps[1]");
+            Assert.That(aPlane.SignedDistanceTo(ps[2]), Is.EqualTo(0).Within(tolerance), "ps[2]");
+        }
+
+        [TestCase(ZeroPoint, X, Y, XY, Z, "0,0,+1")] //all 4 points fall on the plane z=+1
+        [TestCase(ZeroPoint, X, Y, XY, Z, "0,0,-1")] //all 4 points fall on the plane z=-1
+        [TestCase(ZeroPoint, Y, Z, YZ, X, "+1,0,0")] //all 4 points fall on the plane x=+1
+        [TestCase(ZeroPoint, Y, Z, YZ, X, "-1,0,0")] //all 4 points fall on the plane x=-1
+        [TestCase(ZeroPoint, Z, X, ZX, Y, "0,+1,0")] //all 4 points fall on the plane y=+1
+        [TestCase(ZeroPoint, Z, X, ZX, Y, "0,-1,0")] //all 4 points fall on the plane y=-1
+        public void CreateFittedPlaneFrom_With4PointsOnAxisAlignedPlane(string sV1, string sV2, string sV3, string sV4, string sENormal, string sRootPoint)
+        {
+            var rootPoint = Point3D.Parse(sRootPoint);
+            var ps = new[]
+            {
+                Vector3D.Parse(sV1),
+                Vector3D.Parse(sV2),
+                Vector3D.Parse(sV3),
+                Vector3D.Parse(sV4),
+            }.Select(deltaV => rootPoint + deltaV);
+            var actual = Plane.CreateFittedPlaneFrom(ps);
+
+            var eNormal = UnitVector3D.Parse(sENormal);
+            var eRootPoint = Point3D.Parse(sRootPoint);
+            var expected = new Plane(eNormal, eRootPoint);
+            Assert.That(actual, Is.EqualTo(expected));
+        }
+
+        [TestCase("1,1,9;1,2,14;1,3,20;2,1,11;2,2,17;2,3,23;3,1,15;3,2,20;3,3,26", -21.7240278973, -41.0640090729, 7.3269978417, 1)]
```

Comment:
We need more tests, which cover what happens when we have less than 3 points (2, 1, or none).

Reply:
`Plane.CreateFittedPlaneFrom(points)` is not for the testcases with insufficient points.
`Plane.CreateFittedPlaneFrom(points)` is for over-determined system.
Instead, `Plane.Validate(points)` will do for this purpose.

It goes a detailed implementation, `Plane.Validate(points)` will be something like:
```csharp
public static void Validate(IEnumerable<Point3D> points)
{
    //      [p0x, p0y, p0z]
    //      [p1x, p1y, p1z]
    // mat= [p2x, p2y, p2z]
    //      [p3x, p3y, p3z]
    //      [p4x, p4y, p4z]
    //      [..., ..., ...]

    //this expression will be fixed, according to the discussion
    var mat = CreateMatrix.DenseOfRowVectors(
                points.Select(p => CreateVector.DenseOfArray(new double[] { p.X, p.Y, p.Z })));

    var rankOfMat = mat.Rank();  // the library uses SVD.
    if(rankOfMat <= 0) // DoF=0
    {
        throws new Exception("The same point?");
    }
    if(rankOfMat <= 1) // DoF=1
    {
        throws new Exception("Are they collinear?");
    }

    // do nothing intentionally
    // because there are no problem, we can span 2d-vector-space with given points
}
```



If few (2 or less) points are given, the method `Plane.TryFromPonits()` should return `false`. It is similar to `int.TryParse(string)`'s behavior.

Reply:
I think this is over-engineering the problem. We don't need a `Plane.Validate` method, we only need to check in this "best fit method" that before we dive in and calculate the SVD of A, its row count is more than or equal to 3. Something like `if (A.Rows < 3) { throw ...}` (the syntax might not be 100% perfect)